mayastor/src/bdev/nexus/nexus_child_error_store.rs

```diff
@@ -0,0 +1,306 @@
+use spdk_sys::{spdk_bdev, spdk_bdev_io_type};
+use std::{
+    fmt::{Debug, Display},
+    time::{Duration, Instant},
+};
+
+use crate::subsys::Config;
+
+use crate::bdev::nexus::nexus_io::{io_status, io_type};
+
+use crate::core::{Cores, Reactors};
+
+use crate::bdev::nexus::{
+    nexus_bdev,
+    nexus_bdev::{
+        nexus_lookup,
+        Error::{ChildMissing, ChildMissingErrStore},
+        Nexus,
+    },
+};
+
+use serde::export::{fmt::Error, Formatter};
+
+#[derive(Copy, Clone)]
+pub struct NexusChildErrorRecord {
+    io_offset: u64,
+    io_num_blocks: u64,
+    timestamp: Instant,
+    io_error: i32,
+    io_op: spdk_bdev_io_type,
+}
+
+impl Default for NexusChildErrorRecord {
+    fn default() -> Self {
+        Self {
+            io_op: 0,
+            io_error: 0,
+            io_offset: 0,
+            io_num_blocks: 0,
+            timestamp: Instant::now(), // for lack of another suitable default
```

Comment:
FYI - Because you are using default values for the u64 and i32 variables you could just use their default values. So I think you could do something like
Self {timestamp: Instant::now(), ..Default::default()};

Reply:
Oh OK, didn't know about that. Thanks!